check-param-names: does not check arrow function properties in interfaces

## Expected behavior

`check-param-names` should flag mismatched `@param` names on arrow function properties in TypeScript interfaces, the same way it does for method signatures (added in #1251).

## Actual behavior

Only method signature syntax (`TSMethodSignature`) is checked. Arrow function property syntax (`TSPropertySignature`) is silently skipped.

## ESLint Config

```json
// package.json
{
  "dependencies": {
    "eslint": "^9",
    "eslint-plugin-jsdoc": "^62.5.4",
    "typescript": "^5",
    "typescript-eslint": "^8.55.0"
  },
  "scripts": {
    "lint:check": "eslint"
  },
  "type": "module"
}
```

```js
// eslint.config.js
import parser from '@typescript-eslint/parser'
import jsdoc from 'eslint-plugin-jsdoc'

export default [
  {
    files: ['**/*.ts'],
    languageOptions: { parser },
    plugins: {
      jsdoc
    },
    rules: {
      'jsdoc/check-param-names': 'warn'
    }
  }
]
```

## ESLint sample

```js
// index.d.ts
interface Foo {
  /** @param wrongName ... */
  method(name: string): void         // ✅ TSMethodSignature is checked
  /** @param wrongName ... */
  arrow: (name: string) => void      // ❌ TSPropertySignature is not checked
}
```

(Only one warning instead of the expected two.)

```bash
bun run lint:check
~/testing-jsdoc/index.d.ts
  2:1  warning  Expected @param names to be "name". Got "wrongName"  jsdoc/check-param-names

✖ 1 problem (0 errors, 1 warning)
```

## Environment

- Node version: v25.6.1
- ESLint version: v9.39.2
- `eslint-plugin-jsdoc` version: 62.5.4

--- a/docs/rules/check-param-names.md
+++ b/docs/rules/check-param-names.md
@@ -724,6 +724,14 @@ export function fn(...[type, arg]: FnArgs): void {
   // ...
 }
 // Message: Expected @param name to be "type". Got "arg".
+
+interface Foo {
+  /** @param wrongNameA ... */
+  method(name1: string): void
+  /** @param wrongNameB ... */
+  arrow: (name2: string) => void
+}
+// Message: Expected @param names to be "name1". Got "wrongNameA".
 ````
 
 

--- a/src/jsdocUtils.js
+++ b/src/jsdocUtils.js
@@ -476,11 +476,14 @@ const getFunctionParameterNames = (
     return [];
   }
 
-  return (/** @type {import('@typescript-eslint/types').TSESTree.FunctionDeclaration} */ (
-    functionNode
-  ).params || /** @type {import('@typescript-eslint/types').TSESTree.MethodDefinition} */ (
-    functionNode
-  ).value?.params || []).map((param) => {
+  return (
+    /** @type {import('@typescript-eslint/types').TSESTree.TSFunctionType} */
+    (/** @type {import('@typescript-eslint/types').TSESTree.TSPropertySignature} */ (functionNode)?.typeAnnotation?.typeAnnotation)?.params ||
+    /** @type {import('@typescript-eslint/types').TSESTree.FunctionDeclaration} */ (
+      functionNode
+    ).params || /** @type {import('@typescript-eslint/types').TSESTree.MethodDefinition} */ (
+      functionNode
+    ).value?.params || []).map((param) => {
     return getParamName(param);
   });
 };

--- a/src/rules/checkParamNames.js
+++ b/src/rules/checkParamNames.js
@@ -365,7 +365,7 @@ const validateParameterNamesDeep = (
 const allowedNodes = [
   'ArrowFunctionExpression', 'FunctionDeclaration', 'FunctionExpression', 'TSDeclareFunction',
   // Add this to above defaults
-  'TSMethodSignature',
+  'TSMethodSignature', 'TSPropertySignature',
 ];
 
 export default iterateJsdoc(({

--- a/test/rules/assertions/checkParamNames.js
+++ b/test/rules/assertions/checkParamNames.js
@@ -1318,6 +1318,30 @@ export default /** @type {import('../index.js').TestCases} */ ({
         sourceType: 'module',
       },
     },
+    {
+      code: `
+        interface Foo {
+          /** @param wrongNameA ... */
+          method(name1: string): void
+          /** @param wrongNameB ... */
+          arrow: (name2: string) => void
+        }
+      `,
+      errors: [
+        {
+          line: 3,
+          message: 'Expected @param names to be "name1". Got "wrongNameA".',
+        },
+        {
+          line: 5,
+          message: 'Expected @param names to be "name2". Got "wrongNameB".',
+        },
+      ],
+      languageOptions: {
+        parser: typescriptEslintParser,
+        sourceType: 'module',
+      },
+    },
   ],
   valid: [
     {

